packages/react-dom/src/__tests__/ReactDOMServerIntegrationHooks-test.js

```diff
@@ -910,1104 +901,4 @@ describe('ReactDOMServerHooks', () => {
     );
     expect(container.children[0].textContent).toEqual('0');
   });
-
-  describe('useOpaqueIdentifier', () => {
-    it('generates unique ids for server string render', async () => {
-      function App(props) {
-        const idOne = useOpaqueIdentifier();
-        const idTwo = useOpaqueIdentifier();
-        return (
-          <div>
-            <div aria-labelledby={idOne} />
-            <div id={idOne} />
-            <span aria-labelledby={idTwo} />
-            <span id={idTwo} />
-          </div>
-        );
-      }
-
-      const domNode = await serverRender(<App />);
-      expect(domNode.children.length).toEqual(4);
-      expect(domNode.children[0].getAttribute('aria-labelledby')).toEqual(
-        domNode.children[1].getAttribute('id'),
-      );
-      expect(domNode.children[2].getAttribute('aria-labelledby')).toEqual(
-        domNode.children[3].getAttribute('id'),
-      );
-      expect(domNode.children[0].getAttribute('aria-labelledby')).not.toEqual(
-        domNode.children[2].getAttribute('aria-labelledby'),
-      );
-      expect(
-        domNode.children[0].getAttribute('aria-labelledby'),
-      ).not.toBeNull();
-      expect(
-        domNode.children[2].getAttribute('aria-labelledby'),
-      ).not.toBeNull();
-    });
-
-    it('generates unique ids for server stream render', async () => {
```

Comment:
Do we have this coverage for useId?

Reply:
Yeah there are tests for the all the basic scenarios (siblings, parents, indirections, etc) in this file: https://github.com/facebook/react/blob/main/packages/react-dom/src/__tests__/ReactDOMUseId-test.js. They're necessarily more complex than this one because the generation algorithm is different (useOpaqueIdentifier just used a global counter).

For additional coverage I'm planning to write a fuzz tester.